graphql-dgs-codegen-gradle/src/main/kotlin/com/netflix/graphql/dgs/codegen/gradle/ClientUtilsConventions.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ *
+ *  Copyright 2020 Netflix, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.netflix.graphql.dgs.codegen.gradle
+
+import org.gradle.api.Project
+import org.gradle.api.logging.Logging
+import java.util.*
+
+object ClientUtilsConventions {
+
+    private const val CLIENT_UTILS_ARTIFACT_GROUP = "com.netflix.graphql.dgs.codegen"
+    private const val CLIENT_UTILS_ARTIFACT_NAME = "graphql-dgs-codegen-client-utils"
+    private const val GRADLE_CLASSPATH_CONFIGURATION = "compileClasspath"
+
+    private val logger = Logging.getLogger(ClientUtilsConventions::class.java)
+
+    fun apply(project: Project, optionalCodeUtilsVersion: Optional<String> = Optional.empty()) {
+        clientUtilsArtifact(optionalCodeUtilsVersion).ifPresent { dependencyString ->
+            val implementationClasspath = project.configurations.getByName(GRADLE_CLASSPATH_CONFIGURATION).dependencies
+            implementationClasspath.add(project.dependencies.create(dependencyString))
+            logger.lifecycle("DGS CodeGen added [$dependencyString] to the $GRADLE_CLASSPATH_CONFIGURATION.")
+        }
+    }
+
+//    fun apply(project: Project, optionalCodeUtilsVersion: Optional<String> = Optional.empty()) {
```

Comment:
comments

Reply:
Planning on removing this. The example that is commented intended to only add the dependency of the module if the it was not explicitly present.